stdlib/Statistics/src/Statistics.jl

```diff
@@ -74,6 +74,7 @@ function mean(f, itr)
     return total/count
 end
 mean(f, A::AbstractArray) = sum(f, A) / length(A)
+mean(f, A::AbstractArray; dims = 1) = sum(f, A, dims = dims) / prod(size(A, i) for i in dims)
```

Comment:
This actually overwrites the previous definition. The solution is to do this (as elsewhere in the file):
```julia
mean(f, A::AbstractArray; dims = :) = _mean(f, A, dims)

_mean(f, A::AbstractArray, ::Colon) = sum(f, A) / length(A)
_mean(f, A::AbstractArray, dims) = sum(f, A, dims = dims) / prod(size(A, i) for i in dims)
```